### ISSUE TITLE
Why was this deprecated in favor of 'del'?

### ISSUE THREAD

**Issue by ORESoftware** [peter-vilja/gulp-clean#23]
"Deprecated in favor of https://github.com/gulpjs/gulp/blob/master/docs/recipes/delete-files-folder.md"

require('del') is not an event emitter and is not really a straightforward replacement for this lib, is it?

**Comment by palmerj** [comment]
Yes I've had trouble with the recipe replacement, when the clean task not completing before other task dependency start. I've been using del 2.2 on MacOSX, which is supposed to return a promise. e.g 

```js
// clean-up dist directory
gulp.task('clean', function() {
    return del([options.distDir]);
});
```

Can anyone comments on this?
